src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from "react";
+
+const useLocalStorage = <T>(
+  key: string,
+  state: T
+): [T, React.Dispatch<React.SetStateAction<T>>] => {
```

Comment:
Great usage of generics!!
P.S. Import Dispatch separately - again this will improve readability.